src/components/Benefits/BenefitList.tsx

```diff
@@ -40,7 +45,18 @@ const BenefitList = ({
         data={benefits}
         keyExtractor={(benefit) => benefit.publicId}
         renderItem={({ item }) => (
-          <BenefitTile name={item.name} onPress={onPress} tileStyle={benefitStyle}>
+          <BenefitTile
+            name={item.name}
+            onPress={
+              onPress
+                ? onPress
+                : () => {
+                    setBenefit(item);
+                    setState('benefit');
```

Comment:
Wyciągnąłbym to na zewnątrz do jakiegoś handlera typu `alternativeOnPressHandler`